### PR TITLE
Rewrite offending test

### PR DESCRIPTION
**Pull request by AaronRobinsonMSFT**
Rewrite the offending test and re-enable all related skipped tests.

### CODE REVIEW

**Review by jkoritzinsky — APPROVED** [review]

File: src/tests/Interop/MarshalAPI/IUnknown/IUnknownTest.cs [hunk 1]
```diff
@@ -138,69 +82,31 @@ public void GetObjectForIUnknownTest()
...
             {
                 if (ptr != IntPtr.Zero)
+                {
                     Marshal.Release(ptr);
+                }
             }
         }
     }
 
-    public void GetUniqueObjectForIUnknownTest()
```
Comment (elinor-fung) [inline src/tests/Interop/MarshalAPI/IUnknown/IUnknownTest.cs:155]: Are the removed cases based on them already being in libraries? And I guess it looks like they weren't called here either.

File: src/tests/Interop/MarshalAPI/IUnknown/IUnknownTest.cs [hunk 2]
```diff
@@ -29,22 +29,12 @@ public override string ToString()
 
 public class IUnknownMarshalingTest
 {
-    [DllImport(@"IUnknownNative", CallingConvention = CallingConvention.Cdecl)]
-    private static extern bool Marshal_IUnknown([In]IntPtr ptr);
```
Comment (elinor-fung) [inline src/tests/Interop/MarshalAPI/IUnknown/IUnknownTest.cs:33]: Could this test just go in libraries now that it just tests the public APIs and has no native dependencies?

**Review by elinor-fung — APPROVED** [review]
Assuming this is not going in until the underlying issue for https://github.com/dotnet/runtime/issues/85234 is fixed.

File: src/tests/Interop/MarshalAPI/IUnknown/IUnknownTest.cs (hunk 1 above)
Comment (AaronRobinsonMSFT) [reply src/tests/Interop/MarshalAPI/IUnknown/IUnknownTest.cs:155]: Correct.

File: src/tests/Interop/MarshalAPI/IUnknown/IUnknownTest.cs (hunk 2 above)
Comment (AaronRobinsonMSFT) [reply src/tests/Interop/MarshalAPI/IUnknown/IUnknownTest.cs:33]: They could, but I like having these basic ones in the runtime test suite as opposed to the more extensive ones in libraries.

**Comment by AaronRobinsonMSFT** [comment]
https://github.com/dotnet/runtime/issues/85234 is fixed so merging this.

**Pull request merged**